### DISCUSSION TITLE
16xRelay with ESP12F Onboard - only 8 relay configuration and only 8 relay v HA

### DISCUSSION THREAD

**Discussion by Lokalnet99** [arendst/Tasmota#16768] [Q&A] [answered]
Hello everyone,
I bought a 16 relay board, this is exactly it - (https://templates.blakadder.com/ESP12F_Relay_X16.html)

After compiling Tasmota and uploading to the device I see 16 relays fine and all working and can be controlled from the device web page. See screen. Unfortunately, I only see 8 relays in detailed settings... see screen
The same after adding it to Home Assistant, I can only see 8 relays, the others are not loaded, they are not even visible in detail in the device, see the attached screen.

Can someone please tell me what is wrong??
Sorry for my English

web page working
![rele](https://user-images.githubusercontent.com/61562473/194750136-bab771c8-118e-4815-9f23-8958639d71dc.jpg)

Detail confuguration only 8 relay
![rele2](https://user-images.githubusercontent.com/61562473/194750144-658f0a08-a25d-43df-94b9-8c73e6f73f4e.jpg)

HA only 8 relay
![rele_HA](https://user-images.githubusercontent.com/61562473/194750147-e4147718-055c-4932-9646-93705565283b.jpg)

**Comment by sfromis** [comment, marked as answer]
Tasmota does not fully support more than 8 relays. This is not changed by the "trick" of a device with shift registers presenting more than 8 relays for at least partial support, which does not extend to discovery with HA.

I'd guess that you'd be able to configure the upper 8 relays manually.

**Reply by barbudor** [reply]
Whether it is ESP8266 or ESP32, the number of Friendly names is limited to 8
The discovery message includes no more than those 8 even if more POWER are defined
So as sfromis pointed, for those, manual configuration is the way

**Reply by Lokalnet99** [reply]
Thank you very much for the explanation. Can you please tell me how to set it manually

**Reply by sfromis** [reply]
Not being a user of HA, not really, but the Tasmota docs does have some info and examples, like here:
https://tasmota.github.io/docs/Home-Assistant/#switches

**Reply by Lokalnet99** [reply]
thanks

**Comment by phaethon** [comment]
Is there an open Tasmota issue for support of more than 8 relays? I have a similar issue with a different relay board.

**Reply by sfromis** [reply]
No. It would not be an "issue" (which is for bugs), but a potential enhancement request. Were you not able to config manually with HA?

**Reply by Lokalnet99** [reply]
Unfortunately I was unable to configure it manually.  I tried according to the instructions but it doesn't work.....

**Comment by Sellig31** [comment]
Hello, I have a similar problem, And was wondering if it would work in Matter?

I'll give it a try.
And let you know
